core/src/main/java/com/datastax/oss/driver/internal/core/metadata/schema/parsing/RawColumn.java

```diff
@@ -137,9 +127,9 @@ private RawColumn(
   public int compareTo(@NonNull RawColumn that) {
     // First, order by kind. Then order partition key and clustering columns by position. For
     // other kinds, order by column name.
-    if (this.kind != that.kind) {
+    if (!this.kind.equals(that.kind)) {
       return this.kind.compareTo(that.kind);
-    } else if (kind == Kind.PARTITION_KEY || kind == Kind.CLUSTERING_COLUMN) {
+    } else if (kind.equals(KIND_PARTITION_KEY) || kind.equals(KIND_CLUSTERING_COLUMN)) {
```

Comment:
One slight difference, the previous implementation used `equalsIgnoreCase` when determining what `Kind` should be mapped to.  Here we are using `equals`.  It doesn't look like in the C* metadata that anything other then all lowercase was ever used, so not sure if that was ever needed to begin with.